packages/bitcore-wallet-service/src/lib/common/utils.ts

```diff
@@ -275,12 +276,29 @@ export class Utils {
     return coin == 'bch' ? result.toLegacyAddress() : result.toString();
   }
 
-  static compareNetworks(network1, network2) {
-    network1 = network1?.toLowerCase();
-    network2 = network2?.toLowerCase();
+  static compareNetworks(network1, network2, chain) {
+    network1 = network1 ? this.getNetworkName(chain, network1.toLowerCase()) : null;
+    network2 = network2 ? this.getNetworkName(chain, network2.toLowerCase()) : null;
 
     if (network1 == network2) return true;
     if (Config.allowRegtest && ['testnet', 'regtest'].includes(network1) && ['testnet', 'regtest'].includes(network2)) return true;
```

Comment:
Hmm. We still need to match on network type down here... I wonder if we need a util fn like 
```
function isTestnet(network) {
  return !['regtest', 'mainnet', 'livenet'].includes(network);
}
```

or
```
function getNetworkType(network) {
  if (['mainnet', 'livenet'].includes(network)) {
    return 'mainnet';
  ]
  if (network === 'regtest') {
     return 'regtest';
  }
  return 'testnet';
}
```

Reply:
Good catch. I removed that in err. I feel like the existing getGenericName does the same thing that your second function does.

Reply:
`getGenericName` only returns "testnet" based on Constants.NETWORK_ALIASES. `getGenericName('holesky')` returns "holesky", for example.

Speaking of which, shouldn't `Constants.NETWORK_ALIASES.eth.testnet` be "goerli"?